src/utils/wallet.js

```diff
@@ -328,27 +326,46 @@ class Wallet {
         return null
     }
 
-    async signInWithLedger() {
+    async getLedgerAccountIds() {
         const publicKey = await this.getLedgerPublicKey()
         await setKeyMeta(publicKey, { type: 'ledger' })
-        const accountIds = await getAccountIds(publicKey.toString())
+        return (
+            (await Promise.all(
+                (await getAccountIds(publicKey.toString()))
+                    .map(async (accountId) => 
+                        await this.getAccount(accountId).findAccessKey()
+                            ? accountId
+                            : null
+                    )
+                )
+            )
+            .filter(accountId => accountId)
+        )
+    }
+
+    async addLedgerAccountId(accountId) {
+        const accessKeys =  await this.getAccessKeys(accountId)
+        const localAccessKey = await this.getLocalAccessKey(accountId, accessKeys)
+        return await this.addWalletMetadataAccessKeyIfNeeded(accountId, localAccessKey)
+    }
+
+    async saveAndSelectLedgerAccounts(accounts) {
+        const accountIds = Object.keys(accounts)
+        const newKeyPairs = Object.values(accounts)
 
         for (let i = 0; i < accountIds.length; i++) {
             const accountId = accountIds[i]
-            const accessKeys =  await this.getAccessKeys(accountId)
-            const localAccessKey = await this.getLocalAccessKey(accountId, accessKeys)
-            let newKeyPair = await this.addWalletMetadataAccessKeyIfNeeded(accountId, localAccessKey)
-
+            const newKeyPair = newKeyPairs[i]
             if (i === accountIds.length - 1) {
-                await this.saveAndSelectAccount(accountId, newKeyPair)
+                await wallet.saveAndSelectAccount(accountId, newKeyPair)
```

Comment:
why change from `this.` to `wallet.` here?

Reply:
mistake